From 18741b3bc2284c8d02bf1b772a0d41ea0f1b4a67 Mon Sep 17 00:00:00 2001
From: Karthikeyan Periyasamy <quic_periyasa@quicinc.com>
Date: Sun, 22 Oct 2023 20:18:06 +0530
Subject: [PATCH 5/6] wifi: ath12k: Refactor rxdma replenish to avoid explicit rbm

Currently rxdma replenish require return buffer manager id which is an
unnecessary argument since rxdma use the static return buffer manager id
from the HW param. So to optimize the per packet rx data path CPU cycles,
avoid the explicit unnecessary argument in the rx replenish.

Signed-off-by: Karthikeyan Periyasamy <quic_periyasa@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/dp.c    |  6 ++----
 drivers/net/wireless/ath/ath12k/dp_rx.c | 16 ++++++----------
 drivers/net/wireless/ath/ath12k/dp_rx.h |  3 +--
 3 files changed, 9 insertions(+), 16 deletions(-)

--- a/drivers/net/wireless/ath/ath12k/dp.c
+++ b/drivers/net/wireless/ath/ath12k/dp.c
@@ -1417,8 +1417,7 @@ int ath12k_dp_service_srng(struct ath12k
 		struct ath12k_dp *dp = &ab->dp;
 		struct dp_rxdma_ring *rx_ring = &dp->rx_refill_buf_ring;
 
-		ath12k_dp_rx_bufs_replenish(ab, rx_ring, 0,
-					    ab->hw_params->hal_params->rx_buf_rbm);
+		ath12k_dp_rx_bufs_replenish(ab, rx_ring, 0);
 	}
 
 	/* TODO: Implement handler for other interrupts */
@@ -1638,11 +1637,11 @@ void ath12k_dp_vdev_tx_attach(struct ath
 
 void ath12k_dp_umac_txrx_desc_cleanup(struct ath12k_base *ab)
 {
-	struct ath12k_rx_desc_info *desc_info, *tmp;
+	struct ath12k_rx_desc_info *desc_info;
 	struct ath12k_tx_desc_info *tx_desc_info, *tmp1;
 	struct ath12k_dp *dp = &ab->dp;
 	struct sk_buff *skb;
-	int i;
+	int i, j;
 
 	/* RX Descriptor cleanup */
 	spin_lock_bh(&dp->rx_desc_lock);
@@ -2524,8 +2523,7 @@ int ath12k_dp_rxdma_ring_setup(struct at
 		return ret;
 	}
 
-	ath12k_dp_rx_bufs_replenish(ab, rx_ring, 0,
-				    ab->hw_params->hal_params->rx_buf_rbm);
+	ath12k_dp_rx_bufs_replenish(ab, rx_ring, 0);
 	return 0;
 }
 
--- a/drivers/net/wireless/ath/ath12k/dp_rx.c
+++ b/drivers/net/wireless/ath/ath12k/dp_rx.c
@@ -226,8 +226,7 @@ static int ath12k_dp_purge_mon_ring(stru
 /* Returns number of Rx buffers replenished */
 int ath12k_dp_rx_bufs_replenish(struct ath12k_base *ab,
 				struct dp_rxdma_ring *rx_ring,
-				int req_entries,
-				enum hal_rx_buf_return_buf_manager mgr)
+				int req_entries)
 {
 	struct ath12k_buffer_addr *desc;
 	struct hal_srng *srng;
@@ -238,6 +237,7 @@ int ath12k_dp_rx_bufs_replenish(struct a
 	dma_addr_t paddr;
 	struct ath12k_dp *dp = &ab->dp;
 	struct ath12k_rx_desc_info *rx_desc;
+	enum hal_rx_buf_return_buf_manager mgr = ab->hw_params->hal_params->rx_buf_rbm;
 
 	req_entries = min(req_entries, rx_ring->bufs_max);
 
@@ -387,8 +387,7 @@ static int ath12k_dp_rxdma_ring_buf_setu
 	if ((ringtype == HAL_RXDMA_MONITOR_BUF) || (ringtype == HAL_TX_MONITOR_BUF))
 		ath12k_dp_mon_buf_replenish(ab, rx_ring, num_entries);
 	else
-		ath12k_dp_rx_bufs_replenish(ab, rx_ring, 0,
-					    ab->hw_params->hal_params->rx_buf_rbm);
+		ath12k_dp_rx_bufs_replenish(ab, rx_ring, 0);
 	return 0;
 }
 
@@ -3660,9 +3659,7 @@ try_again:
 
 		rx_ring = &src_ab->dp.rx_refill_buf_ring;
 
-		/* TODO: Move to implicit BM? */
-		ath12k_dp_rx_bufs_replenish(src_ab, rx_ring, num_buffs_reaped[i],
-					    src_ab->hw_params->hal_params->rx_buf_rbm);
+		ath12k_dp_rx_bufs_replenish(src_ab, rx_ring, num_buffs_reaped[i]);
 	}
 
 	ath12k_dp_rx_process_received_packets(ab, napi, &msdu_list,
@@ -4528,8 +4525,7 @@ exit:
 
 		rx_ring = &src_ab->dp.rx_refill_buf_ring;
 
-		ath12k_dp_rx_bufs_replenish(src_ab, rx_ring, num_buffs_reaped[i],
-					    src_ab->hw_params->hal_params->rx_buf_rbm);
+		ath12k_dp_rx_bufs_replenish(src_ab, rx_ring, num_buffs_reaped[i]);
 	}
 
 	return tot_n_bufs_reaped;
@@ -5013,8 +5009,7 @@ int ath12k_dp_rx_process_wbm_err(struct
 
 		rx_ring = &src_ab->dp.rx_refill_buf_ring;
 
-		ath12k_dp_rx_bufs_replenish(src_ab, rx_ring, num_buffs_reaped[i],
-					    src_ab->hw_params->hal_params->rx_buf_rbm);
+		ath12k_dp_rx_bufs_replenish(src_ab, rx_ring, num_buffs_reaped[i]);
 	}
 
 	rcu_read_lock();
--- a/drivers/net/wireless/ath/ath12k/dp_rx.h
+++ b/drivers/net/wireless/ath/ath12k/dp_rx.h
@@ -203,8 +203,7 @@ int ath12k_dp_rx_process(struct ath12k_b
 			 int budget);
 int ath12k_dp_rx_bufs_replenish(struct ath12k_base *ab,
 				struct dp_rxdma_ring *rx_ring,
-				int req_entries,
-				enum hal_rx_buf_return_buf_manager mgr);
+				int req_entries);
 int ath12k_dp_rx_pdev_mon_attach(struct ath12k *ar);
 void ath12k_dp_rx_pdev_mon_detach(struct ath12k_base *ab, const int pdev_idx);
 int ath12k_dp_rx_peer_frag_setup(struct ath12k *ar, struct ath12k_peer *peer,
